community-code/How_to_Monitor_Diagnose_and_Solve_Gradient_Issues_in_Foundation_Models/gradient_norm_tracking_in_pytorch .ipynb

```diff
@@ -0,0 +1,252 @@
+    "    return tokenizer(\n",
+    "        examples[\"sentence1\"],\n",
+    "        examples[\"sentence2\"],\n",
+    "        truncation=True,\n",
+    "        padding=\"longest\",\n",
+    "        return_tensors=\"pt\",\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "tokenized_datasets = dataset.map(tokenize_function, batched=True)\n",
+    "tokenized_datasets.set_format(\n",
+    "    type=\"torch\", columns=[\"input_ids\", \"attention_mask\", \"token_type_ids\", \"label\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4",
+   "metadata": {},
+   "source": [
+    "# Step 2. Prepare the Dataloaders and load the Bert model for sequence classification\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_dataset = (\n",
+    "    tokenized_datasets[\"train\"].shuffle(seed=42).select(range(1000))\n",
+    ")  # Sample for demonstration\n",
+    "\n",
+    "data_collator = DataCollatorWithPadding(tokenizer=tokenizer)\n",
+    "\n",
+    "train_dataloader = DataLoader(train_dataset, shuffle=True, batch_size=8, collate_fn=data_collator)\n",
+    "\n",
+    "tokenized_datasets = dataset.map(tokenize_function, batched=True)\n",
+    "tokenized_datasets.set_format(\n",
+    "    type=\"torch\", columns=[\"input_ids\", \"attention_mask\", \"token_type_ids\", \"label\"]\n",
+    ")\n",
+    "\n",
+    "# model = BertForSequenceClassification.from_pretrained('bert-base-uncased')\n",
+    "# # Move model to device\n",
+    "# model.to(device)\n",
+    "\n",
+    "from transformers import BertConfig, BertForSequenceClassification\n",
+    "\n",
+    "# Random BERT config with no LayerNorm or Dropout\n",
+    "config = BertConfig(\n",
+    "    hidden_dropout_prob=0.0,\n",
+    "    attention_probs_dropout_prob=0.0,\n",
+    "    num_hidden_layers=12,\n",
+    "    num_attention_heads=12,\n",
+    "    hidden_size=768,\n",
+    "    intermediate_size=3072,\n",
+    "    layer_norm_eps=1e-12,\n",
+    "    output_attentions=False,\n",
+    "    output_hidden_states=False,\n",
+    "    num_labels=2,\n",
+    ")\n",
+    "\n",
+    "\n",
+    "# config = BertConfig(hidden_dropout_prob=0.5)  # Add more dropout\n",
+    "model = BertForSequenceClassification(config)  # Not pretrained\n",
+    "model.to(device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6",
+   "metadata": {},
+   "source": [
+    "# Step 3. Initialize Neptune for Logging\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = Run(\n",
+    "    api_token=\"YOUR_API_TOKEN\",  # replace with your Neptune API token\n",
+    "    project=\"YOUR_WORKSPACE/YOUR_PROJECT\",  # replace with your workspace and project name\n",
+    "    experiment_name=\"gradient_tracking_lr=1\",\n",
+    ")\n",
+    "\n",
+    "run.log_configs(\n",
```

Comment:
If using a global `configs` dict, you can just pass the dict to `log_configs()`.

```suggestion
    "run.log_configs(configs)\n",
```